### PR TITLE
Workflow code that run topmodel within ngen Framework

### PR DESCRIPTION
**Pull request by stcui007**
This PR provides a workflow code that incorporate topmodel into the ngen framework by running automated ngen test with topmodel commits.

## Additions

.github/workflows/ngen_integration.yaml
data/realization_config_topmodel_ngenCI.json

## Removals

-

## Changes

-

## Testing

Passded all automated testing

## Todos

-

## Checklist

- [x ] PR has an informative and human-readable title
- [x ] Changes are limited to a single goal (no scope creep)
- [x ] Code can be automatically merged (no conflicts)
- [x ] Code follows project standards (link if applicable)
- [x ] Passes all existing automated tests
- [x ] Any _change_ in functionality is tested
- [x ] New functions are documented (with a description, list of inputs, and expected output)
- [x ] Placeholder code is flagged / future todos are captured in comments
- [ ] Visually tested in supported browsers and devices (see checklist below :point_down:)
- [ ] Project documentation has been updated (including the "Unreleased" section of the CHANGELOG)
- [x ] Reviewers requested with the [Reviewers tool](https://help.github.com/articles/requesting-a-pull-request-review/) :arrow_right:

### Target Environment support

- [ ] Windows
- [x ] Linux
- [ ] Browser

### Accessibility

- [ ] Keyboard friendly
- [ ] Screen reader friendly

### Other

- [ ] Is useable without CSS
- [ ] Is useable without JS
- [ ] Flexible from small to large screens
- [ ] No linting errors or warnings
- [ ] JavaScript tests are passing

### CODE REVIEW

File: .github/workflows/ngen_integration.yaml [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+      - name: Build Topmodel
+        id: submod_build_4
+        uses: ./.github/actions/ngen-submod-build
+        with:
+          mod-dir: " extern/topmodel/"
+          targets: "topmodelbmi"
+
+      - name: Run Ngen Test
+        run: |
+          # Run ngen with topmodel with CFE
```
Comment (madMatchstick) [inline .github/workflows/ngen_integration.yaml:110]: Again, this [realization](https://github.com/NOAA-OWP/ngen/blob/master/data/example_bmi_multi_realization_config.json) couples `noahowp` & `cfe`. 

Is a `topmodel` & `CFE` coupling even sensible/common?  @SnowHydrology

It's probably best to throw `PET` somewhere in the mix.  The lib is being built, but not currently used. @stcui007 - I believe I have a `bmi_multi_pet_topmodel` realization around somewhere if that helps. -thanks again.

**Review by madMatchstick — CHANGES_REQUESTED** [review]

File: .github/workflows/ngen_integration.yaml [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+          mod-dir: "extern/iso_c_fortran_bmi/"
+
+      - name: Build CFE
+        id: submod_build_3
+        uses: ./.github/actions/ngen-submod-build
+        with:
+          mod-dir: "extern/cfe/"
+          targets: "cfebmi"
+
+      - name: Build PET
```
[the remark this answers is not in the document]
Comment (stcui007) [reply .github/workflows/ngen_integration.yaml:69]: No, it is not so can be removed.

File: .github/workflows/ngen_integration.yaml (hunk 1 above)
Comment (stcui007) [reply .github/workflows/ngen_integration.yaml:110]: Tha comment is incorrect. I think it is either topmodel or CFE.

Comment (stcui007) [reply .github/workflows/ngen_integration.yaml:110]: Could try to see if you can find your realization file coupling PET with Topmodel? That would save me time if there are intricacies involved. Thanks.

Comment (madMatchstick) [reply .github/workflows/ngen_integration.yaml:110]: @stcui007 - See pr I submitted to [your fork](https://github.com/stcui007/topmodel/pull/1).  Once that is (squashed &) merged, this pr here should be good to go. -thanks!

**Review by madMatchstick — APPROVED** [review]

**Pull request merged**